src/goPackages.ts

```diff
@@ -2,21 +2,32 @@ import vscode = require('vscode');
 import cp = require('child_process');
 import path = require('path');
 import { getGoRuntimePath, getCurrentGoWorkspaceFromGOPATH, fixDriveCasingInWindows } from './goPath';
-import { isVendorSupported, getCurrentGoPath, getToolsEnvVars, getGoVersion, getBinPath, SemVersion, sendTelemetryEvent } from './util';
+import { isVendorSupported, getCurrentGoPath, getToolsEnvVars, getGoVersion, getBinPath, SemVersion, sendTelemetryEvent, getWorkspaceFolderPath } from './util';
 import { promptForMissingTool, promptForUpdatingTool } from './goInstallTools';
 
 type GopkgsDone = (res: Map<string, string>) => void;
-let allPkgsCache: Map<string, string>;
-let allPkgsLastHit: number;
-let gopkgsRunning: boolean = false;
+interface Cache {
+	entry: Map<string, string>;
+	lastHit: number;
+}
+
 let gopkgsNotified: boolean = false;
-let gopkgsSubscriptions: GopkgsDone[] = [];
 let cacheTimeout: number = 5000;
 
-function gopkgs(): Promise<Map<string, string>> {
+let gopkgsSubscriptions: Map<string, GopkgsDone[]> = new Map<string, GopkgsDone[]>();
+let gopkgsRunning: Set<string> = new Set<string>();
+
+let allPkgsCache: Map<string, Cache> = new Map<string, Cache>();
+
+function gopkgs(workDir?: string): Promise<Map<string, string>> {
 	let t0 = Date.now();
 	return new Promise<Map<string, string>>((resolve, reject) => {
-		const cmd = cp.spawn(getBinPath('gopkgs'), ['-format', '{{.Name}};{{.ImportPath}}'], { env: getToolsEnvVars() });
+		const args = ['-format', '{{.Name}};{{.ImportPath}}'];
+		if (workDir) {
+			args.push('-workDir', workDir);
```

Comment:
This will fail for users who don't have the latest `gopkgs` i.e they have a version where `workDir` is not a supported.
You'll have to something similar to https://github.com/Microsoft/vscode-go/blob/0.6.79/src/goFormat.ts#L33 to prompt the user to update their tool, and fallback to not using the `workDir`